.github/workflows/lint.yml

```diff
@@ -32,5 +32,8 @@ jobs:
       - name: Format Check
         run: cargo fmt -- --check
 
-      - name: Check
+      - name: Check (embedded-db)
```

Comment:
Confused here too. Won't `--all-features` + `--all-targets` lint everything? If so, why run clippy a second time?

Reply:
all-features would enable the "embedded-db" feature and so this would disable any postgres code. This is because we have #[cfg(not(feature = "embedded-db")) for the postgres code. We are doing lint check second time to see that postgres lint doesn't have any error. So one clippy run would check that sqlite lint and other one checks the postgres. The features here don't add the functionality but rather deactivate one and activate the other.

Reply:
Humm is this not addressed by having sqlite in a separate package? `cargo lint --workspace` would hit both the package w/ `embedded-db` and the other sequencer binary that does not enable that feature.

Reply:
its not the binary we are checking. We are checking the sequencer library with embedded-db code activated and postgres disabled and vice versa